### DISCUSSION TITLE
How to list all weekly notes for a given month

### DISCUSSION THREAD

**Discussion by kenanmike** [SilentVoid13/Templater#516] [Help]
I’ve been loving getting to do my regular (daily, weekly, monthly, yearly) reviews through Templater. I’ll list how I’m doing those at the bottom, but how can get a monthly review note to list all the weekly notes for that month? That’s the one part of my regular review workflow I haven’t been able to figure out. 

# Daily Note
## Link to previous day’s note (using tp.file.title so I can populate them all correctly for future dates)
[[<% tp.date.now("YYYY-MM-DD", -1, tp.file.title, "YYYY-MM-DD") %>|⬅]]

## Link to the parent weekly note 
WEEK - [[<% tp.date.now("GGGG-[W]WW", 0, tp.file.title, "YYYY-MM-DD") %>]]

# List all Daily Notes for any given Monthly Note

<%* const currentMonthDates = new Array(
moment(tp.file.title,'YYYY-MM').daysInMonth()).fill(null).map(
(x, i) => moment(tp.file.title,'YYYY-MM').startOf('month').add(i,'days').format("- [[[]YYYY-MM-DD[#]YYYY-MM-DD[|]DD/MM [⦗]ddd[.⦘]]]")).join('\r\n') %>
<%* tR +=  currentMonthDates %>

# List all Monthly Notes for any given Yearly Note - 
<% Array.apply(0, Array(12)).map(function(_,i){return moment(tp.file.title,'YYYY').month(i).format("- [[[]YYYY-MM[#]YYYY-MM[|]MM - MMMM[]]]")}).join('\r\n') %>

**Comment by joebuhlig** [comment]
Try this:

```
<%* const currentDates = new Array(Math.ceil(moment.duration(moment(tp.file.title,'YYYY-MM').daysInMonth(), "days").asWeeks())).fill(null).map((x, i) => moment(tp.file.title,'YYYY-MM').startOf('month').add(i,'weeks').format("[[YYYY-[W]ww[]]]")).join('\r\n') %>
<%* tR += currentDates %>
```

**Comment by I-Pch** [comment]
Hi @kenanmike 😊 

As an alternative to the previous suggestion (😇) ... 
For listing all the `ISO weeks` in a month (for a monthly note where the title is formatted as `YYYY-MM`) I have this template : 
```javascript
<%*
let start_of_month = moment(tp.file.title,'YYYY-MM').startOf('month').startOf('IsoWeek');
let end_of_month = moment(tp.file.title,'YYYY-MM').endOf('month').startOf('IsoWeek'); 
let weeks_in_month = end_of_month.diff(start_of_month, 'weeks') + 1;
tR += Array(weeks_in_month).fill(null).map((x, i) => moment(tp.file.title,'YYYY-MM').startOf('month').startOf('IsoWeek').add(i, 'w').format("- [[[]GGGG-[W]WW[#]GGGG-[W]WW[|]GGGG[-W]WW[]]]")).join('\r\n') 
%>
```
Which returns something like this : 
<img width="233" alt="ISO Weeks per month" src="https://user-images.githubusercontent.com/77828085/151415400-6f28c9d4-0fbd-43a2-9f84-0094da22b430.png">

**Comment by foenixp** [comment]
Hi, is it possible to embed each week of a certain month into that monthly note?

**Reply by I-Pch** [reply]
Normally, yes @foenixp 😊 !

Could you try this and see if this works for you 😇 ? (I'm kind of AFK at the moment so I'm not able to test)  

```JS
<%*
let start_of_month = moment(tp.file.title,'YYYY-MM').startOf('month').startOf('IsoWeek');
let end_of_month = moment(tp.file.title,'YYYY-MM').endOf('month').startOf('IsoWeek'); 
let weeks_in_month = end_of_month.diff(start_of_month, 'weeks') + 1;
tR += Array(weeks_in_month).fill(null).map((x, i) => moment(tp.file.title,'YYYY-MM').startOf('month').startOf('IsoWeek').add(i, 'w').format("![[[]GGGG-[W]WW[#]GGGG-[W]WW[]]]")).join('\r\n') 
%>
```

All I did was changing the format of the weekly links in the `.format(...)` to get embeds instead of a bullet list 😊 ...

**Reply by foenixp** [reply]
It works! Turned out I have to delete the code above for this code to work. Thank you very much! 🥰🥰🥰

**Reply by I-Pch** [reply]
Ah ah 😊 ! Glad to know you've found the solution I was going to suggest you to try 😉 !

With Templater once a variable/constant is declared anywhere in your template, you can re-use it multiple times in other bits of template 😊 ... or you need to use a different name for the constant/variable for it to work 😊 .

As here I just copy/pasted the whole template and modified it, if you had both bits of codes in your actual template, it means that the `start_of_month`,  `end_of_month` and `weeks_in_month` were declared twice leading to the templater failure 😉 ...

**Reply by foenixp** [reply]
Thanks again!
